### PR TITLE
Code Quality: Improved the way packages are managed in the codebase

### LINKED ISSUE #16430: Code Quality: Improve the way packages are managed in the codebase
### Description

Currently, local NuGet Packages, Link Libraries and Windows Metadata files are managed in a very messy way:

- They are stored in `src/Files.App/nupkgs` which is not a PascalCase folder like others
> Preferably, they should be stored in a place like `pkgs/*` or `src/Packages/*` etc.

- ~~App extension files (link libraries) are stored in the root `src/Files.App` directory.~~
> src/Files.App/*.dll
> 
> ~~This is a bad idea as it clutters the codebase when they could more neatly be organised in the same directory as the NuGet package store, or inside another folder altogether.~~

- The NuGet Gallery server entry is duplicated in the NuGet package manager
> NuGet packages for Solution...
> 
> This is because the NuGet Gallery endpoint is also listed inside Files's `nuget.config` file which it _should not_ as this entry is already provided automatically by Visual Studio.

### Concerned code

- https://github.com/files-community/Files/blob/a4641ff9bcc5c6224982fddc6e2f28d798185bac/nuget.config#L4
- https://github.com/files-community/Files/blob/a4641ff9bcc5c6224982fddc6e2f28d798185bac/nuget.config#L5
- ~~`src/Files.App/*.dll`~~
- `src/Files.App/nupkgs/`

### Gains

- Better codebase structure
- Less confusion when choosing a package source
- Central store for NuGet Packages, Link Libraries and Windows Metadata files

### Requirements

This should be removed.
```diff
-         <add key="NuGet official package source" value="https://api.nuget.org/v3/index.json" />
```

---

This should be changed to a different pattern that matches with the rest of the codebase.

```diff
-         <add key="Local Packages" value="src/Files.App/nupkgs" />
+         <add key="Local Packages" value="src/Files.App/Packages" />
```
```diff
-         <add key="Local Packages" value="src/Files.App/nupkgs" />
+         <add key="Local Packages" value="src/Packages" />
```
```diff
-         <add key="Local Packages" value="src/Files.App/nupkgs" />
+         <add key="Local Packages" value="nupkgs" />
```
```diff
-         <add key="Local Packages" value="src/Files.App/nupkgs" />
+         <add key="Local Packages" value="pkgs" />
```

---

~~The `dll` files should also be migrated to the same or a similar store.~~

### Comments

N/A

### PR DESCRIPTION
**Pull request by Lamparter**
**Resolved / Related Issues**

- Fixes #16430

**Steps used to test these changes**

N/A

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
     <packageSources>
-        <add key="NuGet official package source" value="https://api.nuget.org/v3/index.json" />
-        <add key="Local Packages" value="src/Files.App/nupkgs/" />
+        <add key="Project Packages" value="src/Files.App/Assets/Libraries/" />
```
Comment (0x5bfa) [inline nuget.config:4]:
```suggestion
        <add key="Local Packages" value="src/Files.App/Assets/Libraries/" />
```

Comment (Lamparter) [reply nuget.config:4]: "Local Packages" is also the name of the Office offline packages store.

Comment (0x5bfa) [reply nuget.config:4]: Do we use that?

File: src/Files.App/Assets/Libraries/SevenZipSharp.1.0.2.nupkg
Comment (0x5bfa) [inline src/Files.App/Assets/Libraries/SevenZipSharp.1.0.2.nupkg:1]: Also check if seven zip sharp still works by trying to compress or decompress.

Comment (Lamparter) [reply src/Files.App/Assets/Libraries/SevenZipSharp.1.0.2.nupkg:1]: This shouldn't make a difference; unfortunately I can't test it.

**Comment by 0x5bfa** [comment]
LGTM.

**Review by 0x5bfa — APPROVED** [review]
Looks good.

File: src/Files.App/Assets/Libraries/SevenZipSharp.1.0.2.nupkg
Comment (0x5bfa) [reply src/Files.App/Assets/Libraries/SevenZipSharp.1.0.2.nupkg:1]: You're true.

File: nuget.config (hunk 1 above)
Comment (0x5bfa) [reply nuget.config:4]: I see. Btw we should distribute through NuGet.org tho, this way is way much easier to update package.

**Pull request merged**